src/main/java/spark/route/SimpleRouteMatcher.java

```diff
@@ -164,24 +164,19 @@ private boolean routeWithGivenAcceptType(String bestMatch) {
         return matchSet;
     }
 
-    // TODO: I believe this feature has impacted performance. Optimization?
+    // TODO: I believe this feature has impacted performance. Optimization? Optimization?
     private RouteEntry findTargetWithGivenAcceptType(List<RouteEntry> routeMatches, String acceptType) {
-        if (acceptType != null && routeMatches.size() > 0) {
+
+        if (routeMatches.isEmpty()) {
+            return null;
```

Comment:
can "routeMatches" can be null? if spark has apache commons use CollectionUtils.isEmpty() else, just check list is null too